interfaces/prompting/errors/errors.go

```diff
@@ -0,0 +1,242 @@
+func invalidInterfaceError(unsupported string, supported []string) *UnsupportedValueError {
+	return &UnsupportedValueError{
+		Field:       "interface",
+		Msg:         fmt.Sprintf("invalid interface: %q", unsupported),
+		Unsupported: unsupported,
+		Supported:   supported,
+	}
+}
+
+func invalidPermissionsError(iface string, unsupported []string, supported []string) *UnsupportedValueError {
+	return &UnsupportedValueError{
+		Field:       "permissions",
+		Msg:         fmt.Sprintf("invalid permissions for %s interface: %v", iface, unsupported),
+		Unsupported: unsupported,
+		Supported:   supported,
+	}
+}
+
+func permissionsListEmptyError(iface string, supported []string) *UnsupportedValueError {
+	return &UnsupportedValueError{
+		Field:       "permissions",
+		Msg:         fmt.Sprintf("invalid permissions for %s interface: permissions list empty", iface),
+		Unsupported: []string{},
+		Supported:   supported,
+	}
+}
+
+// Marker for UnsupportedValueError, should never be returned as an actual
+// error value.
+var ErrUnsupportedValue = errors.New("unsupported value")
+
+// UnsupportedValueError is a wrapper for errors about a field having an
+// unsupported value when there is a fixed set of supported values.
+type UnsupportedValueError struct {
+	Field       string
+	Msg         string
+	Unsupported interface{} // either string or []string
+	Supported   []string
+}
+
+func (e *UnsupportedValueError) Error() string {
+	return e.Msg
+}
+
+func (e *UnsupportedValueError) Is(target error) bool {
+	return target == ErrUnsupportedValue
+}
+
+// Parse errors, which are built from the ParseError struct
+
+func invalidDurationError(invalid string, reason string) *ParseError {
+	return &ParseError{
+		Field:   "duration",
+		Msg:     fmt.Sprintf("invalid duration: %s: %q", reason, invalid),
+		Invalid: invalid,
+	}
+}
+
+func invalidExpirationError(invalid time.Time, reason string) *ParseError {
+	timeStr := invalid.Format(time.RFC3339Nano)
+	return &ParseError{
+		Field:   "expiration",
+		Msg:     fmt.Sprintf("invalid expiration: %s: %q", reason, timeStr),
+		Invalid: timeStr,
+	}
+}
+
+func invalidPathPatternError(invalid string, reason string) *ParseError {
+	return &ParseError{
+		Field:   "path-pattern",
+		Msg:     fmt.Sprintf("invalid path pattern: %s: %q", reason, invalid),
+		Invalid: invalid,
+	}
+}
+
+// Marker for ParseError, should never be returned as an actual error value.
+var ErrParseError = errors.New("parse error")
+
+// ParseError is a wrapper for errors about a field whose value is malformed.
+type ParseError struct {
+	Field   string
+	Msg     string
+	Invalid string
+}
+
+func (e *ParseError) Error() string {
+	return e.Msg
+}
+
+func (e *ParseError) Is(target error) bool {
+	return target == ErrParseError
+}
+
+// Validation errors, which are all uniquely defined here
+
+// RequestedPathNotMatchedError stores a path pattern from a reply which doesn't
+// match the requested path.
+type RequestedPathNotMatchedError struct {
```

Comment:
to be in sync with other errors this could use:

```
func NewRequestedPathNotMatchedError(received, requested string) *RequestedPathNotMatchedError {
    return &RequestedPathNotMatchedError{
        Received: received,
        Requested: requested,
    }
}
```